### PR TITLE
Plugins: Fix catalog permissions for org and server admins

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
Due to the Plugin Catalog replacing the current plugin list (when the feature flag is enabled) we need to align the different levels of admin and the features they can access. 

This PR gives the following plugin catalog permissions for each level of admin:

| Org Admin | Server Admin | Permissions |
| --- | --- | --- |
| ✅ | ✅ | full catalog abilities |
| ✅ | 🚫 | <ul><li>Can access plugin catalog</li><li>Can configure app plugins</li><li>Cannot install/uninstall/update plugins</li></ul> |
| 🚫 | ✅ | <ul><li>Can access plugin catalog</li><li>Cannot configure app plugins</li><li>Can install/uninstall/update plugins</li></ul> |

**Which issue(s) this PR fixes**:

Related #36229

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/plugins/admin/hooks/usePluginDetails.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import { getCatalogPluginDetails } from '../helpers';
 import { ActionTypes, PluginDetailsActions, PluginDetailsState } from '../types';
 
 const defaultTabs = [{ label: 'Overview' }, { label: 'Version history' }];
 
 const initialState = {
   hasInstalledPanel: false,
   hasUpdate: false,
-  isAdmin: isGrafanaAdmin(),
+  isAdmin: contextSrv.hasRole('Admin'),
```
Comment (wbrowne) [inline public/app/features/plugins/admin/hooks/usePluginDetails.tsx:14]: Probably safest to name this `isOrgAdmin` - WDYT?

File: public/app/features/plugins/admin/components/InstallControls.tsx [hunk 2]
```diff
@@ -94,6 +94,11 @@ export const InstallControls = ({ plugin, isInflight, hasUpdate, isInstalled, ha
     );
   }
 
+  if (!hasPermission) {
+    const message = `You need server admin privileges to ${isInstalled ? 'uninstall' : 'install'} this plugin.`;
```
Comment (wbrowne) [inline public/app/features/plugins/admin/components/InstallControls.tsx:98]: Is it possible to leave a bit more space between the text and button?

image

**Review by wbrowne — APPROVED** [review]
LGTM 🚀

File: public/app/features/plugins/admin/components/InstallControls.tsx (hunk 2 above)
Comment (jackw) [reply public/app/features/plugins/admin/components/InstallControls.tsx:98]: You still see the button? I moved this logic above the button rendering so they don't show any more. 🤔

**Comment by wbrowne** [comment]
Latest commit looks good 👍

**Pull request merged**